packages/php-wasm/universal/src/lib/universal-php.ts

```diff
@@ -439,3 +439,11 @@ export interface RmDirOptions {
 	 */
 	recursive?: boolean;
 }
+
+export interface ListFilesOptions {
+	/**
+	 * If true, prepend given folder path to all file names.
+	 * Default: false.
+	 */
+	includePath: boolean;
```

Comment:
Maybe `absolutePaths`?

Reply:
With the current naive implementation, it just prepends the given path (first argument of function `listFiles`) to every file name. The path may be relative or absolute.

```ts
return files.map((name: string) => `${path}/${name}`);
```

Is there a way to resolve a relative path to absolute?

If so, returning absolute paths to the files would be better, I think, in case the current working directory changes.

Node.js has [`FS.realpath`](https://nodejs.org/api/fs.html#fspromisesrealpathpath-options). (I wish it were properly cased, `realPath`.) For Emscripten, looks like the closest is [`FS.lookupPath`](https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.lookupPath). (Better `lookUpPath`?) They seem to have slightly different behavior, guessing from the available options for each.

Maybe we create a `PHP.resolvePath` function that uses these under the hood?

(I wonder how best to detect which `FS` is being used.. Aha, there's a variable `currentJsRuntime` that return `NODE`, `WEB`, or `WORKER`. Hmm, maybe I'm misunderstanding.. Is `this[__private__dont__use].FS` always the one provided by Emscripten, or does it use `node:fs` module on server side? Aah OK, from the documentation page titled [PHP Filesystem](https://wordpress.github.io/wordpress-playground/docs/architecture/wasm-php-filesystem), I see that it's always Emscripten's `FS`.)

So then, using `FS.lookupPath` to get absolute paths in `listFiles`..

```ts
return files.map(
  (name: string) => `${ FS.lookupPath(path).path }/${name}`
);
```

That looks expensive, resolving (possibly the same) path for every file. I hope it's cached internally.

Reply:
I agree the option name `includePath` is ambiguous in meaning. If we're to keep the current behavior, it should be called `prependPath`. If we will resolve given path to absolute, then `absolutePaths` makes sense.

Reply:
Yeah you're spot-on here, `absolutePath` looks like a lot of trouble. `prependPath` sounds great 👍

Reply:
Actually I see now that the absolute path lookup would only need to happen once, since it's always the same for all file paths under that directory.

```ts
const absolutePath = FS.lookupPath(path).path
return files.map((name: string) => `${ absolutePath }/${name}`);
```

So an option `absolutePaths` could be implemented this way.

Elsewhere in `BasePHP`, I found a usage of `listFiles` like this:

``ts
this.listFiles(path).forEach((file) => {
	const filePath = `${path}/${file}`;
	...
})
``

https://github.com/WordPress/wordpress-playground/blob/01357b94cdb0d13f26823bd345d4a4887e06a983/packages/php-wasm/universal/src/lib/base-php.ts#L496-L497

With the new option, it would look like:

``ts
this.listFiles(path, { prependPath: true }).forEach((filePath) => {
	...
})
``

It's not that much shorter than the original, which may even be better without the indirection (of having to know that this option exists) - but at the cost of repeating this pattern:

```ts
const filePath = `${path}/${file}`;
```

Well, but this pattern is so common that every person who uses `listFiles` will likely need it. OK, so I'll name the option `prependPath`.